sources/iTermNotificationController.m

```diff
@@ -84,6 +85,53 @@ - (BOOL)notify:(NSString *)title
                       sticky:NO];
 }
 
+- (BOOL)notifyRich:(NSString *)title
+      withSubtitle:(NSString *)subtitle
+   withDescription:(NSString *)description
+         withImage:(NSString *)image
+         withSound:(NSString *)sound
+       windowIndex:(int)windowIndex
+          tabIndex:(int)tabIndex
+         viewIndex:(int)viewIndex {
+    DLog(@"Notify title=%@ description=%@ window=%@ tab=%@ view=%@",
+            title, description, @(windowIndex), @(tabIndex), @(viewIndex));
+
+    if (description == nil) {
+        DLog(@"notifyRich passed nil description.");
+        return NO;
+    }
+
+    // Send to NSUserDefaults directly
+    NSUserNotification *notification = [[NSUserNotification alloc] init];
+    notification.informativeText = description;
+
+    NSDictionary *context = nil;
+    if (windowIndex >= 0) {
+        context = @{ @"win": @(windowIndex),
+                @"tab": @(tabIndex),
+                @"view": @(viewIndex) };
+    }
+    notification.userInfo = context;
+
+    if (title != nil) {
+        notification.title = title;
+    }
+    if (subtitle != nil) {
+        notification.subtitle = subtitle;
+    }
+    if (image != nil) {
+        NSData *data = [NSData dataWithContentsOfFile:image];
+        notification.contentImage = [[[iTermImage imageWithCompressedData:data] images] firstObject];
+    }
+    if (sound != nil) {
+        notification.soundName = [sound isEqualToString:@"default"] ? NSUserNotificationDefaultSoundName : sound;
```

Comment:
How is sound supposed to work? I see in NSUserNotification.h that it's supposed to be the name of a sound in the app bundle:

```
// The name of the sound file in the resources of the application bundle to play when the notification is delivered. NSUserNotificationDefaultSoundName can be used to play the default Notification Center sound. A value of 'nil' means no sound.
@property (nullable, copy) NSString *soundName;
```

But there are no sounds in the app bundle. I'd be fine with adding a few if they're not too big and would be generally useful, but otherwise we should remove the sound field since it won't work.

Reply:
Ah, I had misunderstood the documentation.  I had assumed it would search system sounds as well.  But if the sound must be bundled with the application, it seems like the `NSUserNotification` API is not general enough to be worth exposing here.